.github/workflows/publish-docker-images.yml

```diff
@@ -0,0 +1,96 @@
+name: Build Agent Images
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        required: false
+        type: string
+        default: 3.8.10
+
+    secrets:
+      SCALYR_PROD_CLOUDTECH_TESTING_WRITE_TOKEN:
+        required: true
+      SCALYR_CLOUDTECH_TESTING_DEV_SCALYR_READ_API_KEY:
+        required: true
+      DOCKER_HUB_USERNAME:
+        required: true
+      DOCKER_HUB_PASSWORD:
+        required: true
+
+env:
+  DOCKER_BUILDKIT: 1
+  # Set this variable to tell the agent build code that it runs in CI/CD and it needs to use caching.
+  AGENT_BUILD_IN_CICD: "1"
+
+concurrency: publish-release
+
+jobs:
+  publish-temp-release:
+    name: Publish Docker Image to a temporary tag.
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        image-type: [ "docker-json", "docker-syslog", "docker-api", "k8s" ]
+        image-distro-name: [ "buster", "alpine" ]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:qemu-v6.1.0
+          platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+
+      - name: Perform the build of the base docker image in the deployment.
+        uses: ./.github/actions/perform-deployment
+        with:
+          deployment-name: ${{ matrix.image-type }}-${{ matrix.image-distro-name }}
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+      - name: Push image using the a temporary tag.
+        run: |
+          # Push result docker image with a special tag "_temp_release".
+          # The next job will "re-tag" that image to an appropriate one.
+          # That is done to guarantee that the real, production tag only pushed when all job are succeeded.
+          python3 build_package_new.py "${{ matrix.image-type }}-${{ matrix.image-distro-name }}" \
+            --push \
+            --user "${{ secrets.DOCKER_HUB_USERNAME }}" \
+            --tag "_temp_release-${{ matrix.image-distro-name }}"
```

Comment:
Once we re-tag the image with the final tag, will this temporary tag be removed or?

Reply:
Should it be removed? I just thought it's ok to leave it until the next time.

Reply:
I'm fine either way, I also don't see a problem if we leave it there.